Fixing index error after close files and clear `_log_destination`

hunterhector opened the pull request:
This PR fixes https://github.com/asyml/texar-pytorch/issues/322

## The fix
At the time of `_open_files`, we check whether the length of the destination is zero, and fill in the List if needed.

## The test
In the executor tests, we called `test` after `train` in all the tests. In this way, we ensure the `_files_opened` to have both `True` and `False` value, to allow us to test both branches in the `_open_files` function.

huzecong commented:
For some reason adding `executor.test()` doesn't raise exceptions when I test locally without applying the fix... Strange.

This fix definitely works, however I think the easier thing to do is to simply not set `_log_destinations` to `[]`. Since we maintain `_files_opened`, we don't need to check whether `_log_destinations` has length 0.

hunterhector commented:
> For some reason adding `executor.test()` doesn't raise exceptions when I test locally without applying the fix... Strange.

You are right indeed, to make the bug appear we need to add a file log in the log_destination parameter. I have added that in the newer commit.

> This fix definitely works, however I think the easier thing to do is to simply not set `_log_destinations` to `[]`. Since we maintain `_files_opened`, we don't need to check whether `_log_destinations` has length 0.

Added in the newer commit, see if that works.
